src/oop/programs_with_multiple_classes/hard/q2/MainTest.java

```diff
@@ -0,0 +1,130 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private ObjectTest person;
+    private ObjectTest blanket;
+
+    private final String height = "height";
+    private final String size = "size";
+    private final String threadCount = "threadCount";
+
+    private final String personLc = "person";
+    private final String personUc = "Person";
+    private final String blanketLc = "blanket";
+    private final String blanketUc = "Blanket";
+
+    @BeforeEach
+    public void Setup() {
+        String personClassString = "oop.programs_with_multiple_classes.hard.q2." + personUc;
+        String blanketClassString = "oop.programs_with_multiple_classes.hard.q2." + blanketUc;
+        person = new ObjectTest(personClassString);
+        blanket = new ObjectTest(blanketClassString);
+    }
+
+    /**
+     * Person Tests
+     **/
+    @Test
+    public void personClassHasCorrectFields() {
+        String missingFieldMessage = "Your " + personLc + " class is missing a required field.";
+        String incorrectFieldModifierMessage = "A field in your " + personLc + " class has the wrong visibility modifier.";
+        assertTrue(person.hasField(height, double.class), missingFieldMessage);
+        assertTrue(person.hasModifier(height, "private"), incorrectFieldModifierMessage);
+    }
+
+    @Test
+    public void personClassHasCorrectConstructors() {
+        String missingConstructorMessage = "Your " + personLc + " class does not have a required constructor.";
+        String incorrectConstructorModifierMessage =
+                "One of the constructors in your " + personLc + " class does not have the correct visibility modifier.";
+        Class<?>[] constructorClasses = new Class[]{
+                double.class
+        };
+        assertTrue(person.hasConstructor(constructorClasses), missingConstructorMessage);
+        assertTrue(person.hasModifier(constructorClasses, "public"), incorrectConstructorModifierMessage);
+    }
+
+    private static Stream<Double> personInputProvider() {
+        return Stream.of(
+                75.0,
+                63.0,
+                55.0,
+                80.0
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("personInputProvider")
+    public void personConstructorInitializesValuesCorrectly(Double height) throws Throwable {
+        Object[][] arguments = new Object[][]{
+                {height, double.class}
+        };
+        Object personInstance = person.createInstance(arguments);
+        _assertEquals(height, person.getFieldValue(personInstance, this.height),
+                "Your " + personLc + " constructor does not correctly initialize the " + this.height + " field.");
+    }
+
+    /**
+     * Blanket Tests
+     **/
+    @Test
+    public void blanketClassHasCorrectFields() {
+        String missingFieldMessage = "Your " + blanketLc + " class is missing a required field.";
```

Comment:
Please apply the same changes I mentioned in the comments above to your blanket tests.